lib/ladybug/energy_model/energy_material.rb

```diff
@@ -0,0 +1,136 @@
+
+require 'json-schema'
+require 'json'
+require 'openstudio'
+
+module Ladybug
+  module EnergyModel
+    class EnergyMaterial < ModelObject
+      attr_reader :errors, :warnings
+
+      def initialize(hash = {})
+        super(hash)
+
+        raise "Incorrect model type '#{@type}'" unless @type == 'EnergyMaterial'
+      end
+
+      def defaults
+        result = {}
+        result[:type] = 'EnergyMaterial'
+        result[:roughness] = @@schema[:definitions][:EnergyMaterial][:properties][:roughness][:default]
+        result[:thermal_absorptance] = @@schema[:definitions][:EnergyMaterial][:properties][:thermal_absorptance][:default]
+        result[:solar_absorptance] = @@schema[:definitions][:EnergyMaterial][:properties][:solar_absorptance][:default]
+        result[:visible_absorptance] = @@schema[:definitions][:EnergyMaterial][:properties][:visible_absorptance][:default]
+        result
+      end
+
+      def name
+        @hash[:name]
+      end
+
+      def name=(new_name)
+        @hash[:name] = new_name
+      end
+
+      def thickness
+        @hash[:thickness]
+      end
+
+      def thickness=(new_thickness)
+        @hash[:thickness] = new_thickness
+      end
+
+      def conductivity
+        @hash[:conductivity]
+      end
+
+      def conductivity=(new_conductivity)
+        @hash[:conductivity] = new_conductivity
+      end
+
+      def density
+        @hash[:density]
+      end
+
+      def density=(new_density)
+        @hash[:density] = new_density
+      end
+
+      def specific_heat
+        @hash[:specific_heat]
+      end
+
+      def specific_heat=(new_specific_heat)
+        @hash[:specific_heat] = new_specific_heat
+      end
+
+      def find_existing_openstudio_object(openstudio_model)
+        object = openstudio_model.getStandardOpaqueMaterialByName(@hash[:name])
+        return object.get if object.is_initialized
+        nil
+      end
+
+      def create_openstudio_object(openstudio_model)
+        openstudio_opaque_material = OpenStudio::Model::StandardOpaqueMaterial.new(openstudio_model)
+        openstudio_opaque_material.setName(@hash[:name])
+        if @hash[:roughness]
+          openstudio_opaque_material.setRoughness(@hash[:roughness])
+        else
+          openstudio_opaque_material.setRoughness(@@schema[:definitions][:EnergyMaterial][:properties][:roughness][:default])
+        end
+        openstudio_opaque_material.setThickness(@hash[:thickness])
+        openstudio_opaque_material.setConductivity(@hash[:conductivity])
+        openstudio_opaque_material.setDensity(@hash[:density])
+        openstudio_opaque_material.setSpecificHeat(@hash[:specific_heat])
+        if @hash[:thermal_absorptance]
+          openstudio_opaque_material.setThermalAbsorptance(@hash[:thermal_absorptance].to_f)
+        else
+          openstudio_opaque_material.setThermalAbsorptance(@@schema[:definitions][:EnergyMaterial][:properties][:thermal_absorptance][:default].to_f)
```

Comment:
This case can't happen because the default value is already set in the constructor right?

Reply:
I guess someone could call the constructor with a hash setting thermal_absorptance to nil

Reply:
The default value would be taken from the schema in case no default is specified in the `openstudio_model` since its not a required property.